src/libcommon/utility/types.h

```diff
@@ -358,4 +358,10 @@ struct ItemType {
         // `LinkTypeUnknown`) and its target doesn't exist.
         IoError ioError{IoErrorSuccess};
 };
+
+typedef enum {
```

Comment:
Use `enum AppStateKey ` instead (see https://en.cppreference.com/w/cpp/language/enum). Sonar Source should give hints as how to proceed in the best way,

Reply:
Following the Sonar advice (and my programming course) it is even better to use an enum class. (commit: https://github.com/Infomaniak/desktop-kDrive/pull/76/commits/3b8cefecd5477d510239ad455ce38b1ced471f34)